parse_filings.py

```diff
@@ -0,0 +1,259 @@
+
+# returns list of all case nums for all prefixes between afterdate and beforedate
+def get_all_case_nums(afterdate: str, beforedate: str):
+    case_num_prefixes = ["J1-CV-20*", "J2-CV-20*", "J3-EV-20*", "J4-CV-20*", "J5-CV-20*"]
+    all_case_nums = []
+    for prefix in case_num_prefixes:
+        prefix_case_nums = fetch_filings(afterdate, beforedate, prefix)
+        all_case_nums += prefix_case_nums
+    return all_case_nums
+
+# drops all rows with case number in case_ids from table table_name - meant to be used for case_detail, disposition, and event tables only (won't work for settings)
+def drop_rows_from_table(table_name: str, case_ids: list):
+    if len(case_ids) == 1:
+        case_ids = str(tuple(case_ids)).replace(",", "")
+    else:
+        case_ids = str(tuple(case_ids))
+
+    conn = sqlite3.connect("cases.db")
+    curs = conn.cursor()
+
+    if table_name == "CASE_DETAIL":
+        query_string = 'delete from "' + table_name + '" where "ID" in ' + case_ids
+        curs.execute(query_string)
+    else:
+        query_string = 'delete from "' + table_name + '" where "CASE_DETAIL_ID" in ' + case_ids
+        curs.execute(query_string)
+
+    conn.commit()
+    curs.close()
+
+# returns list of case nums in sqlite that are still active, removes all rows with those case numbers from case_detail, disposition, and event tables
+def get_old_active_case_nums():
+    conn = sqlite3.connect("cases.db")
+    curs = conn.cursor()
+
+    curs.execute("""SELECT "ID" FROM "CASE_DETAIL" WHERE "STATUS" NOT IN
+                ('Final Disposition', 'Transferred', 'Bankruptcy', 'Judgment Released',
+                'Judgment Satisfied', 'Appealed', 'Final Status', 'Dismissed')""")
+    active_case_nums = [tup[0] for tup in curs.fetchall()]
+    curs.close()
+
+    for table_name in ["CASE_DETAIL", "EVENT", "DISPOSITION"]:
+        drop_rows_from_table(table_name, active_case_nums)
+
+    return active_case_nums
+
+@click.command()
+@click.argument("afterdate", nargs=1)
+@click.argument("beforedate", nargs=1)
+@click.argument("outfile", default="results.json")
+@click.option('--showbrowser / --headless', default=False, help='whether to operate in headless mode or not')
+
+# performs a full 'scraper run' between afterdate and beforedate - gets case details, events, and dispoitions for all case nums between
+# afterdate and beforedate, also updates rows in event/disposition/case_detail table that are still active
+def parse_filings(afterdate, beforedate, outfile, showbrowser=False):
+    # use default firefox browser - use the commented line instead once you move helpers to another file
+    if showbrowser:
+        # fetch_page.driver = webdriver.Firefox()
+        driver = webdriver.Firefox()
+
+    all_case_nums = get_all_case_nums(afterdate, beforedate) + get_old_active_case_nums()
+
+    # save case nums as csv, give to parse_all function - may be a more normal way to do this
+    all_case_nums_df = pd.DataFrame(all_case_nums)
```

Comment:
It seems like the `pandas` import is not really needed, if all you're doing is using it to export CSV.

Reply:
I agree, probably a smarter way to do that - I'll probably change it so it doesn't need a csv and can just feed the list into the hearings scraper